### PR TITLE
PORTAL-869: Add phone numbers endpoints

### PR DESCRIPTION
**Pull request by kanemathers**

### CODE REVIEW

File: telnyx/api_resources/phone_number.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+    def messaging(self, **params):
+        """ Returns the messaging settings for the instantiated phone
+        number.
+        """
+
+        return PhoneNumber.list_messaging(self.id, **params)
+
+
+class VoiceSettings(ListableAPIResource, UpdateableAPIResource):
```
Comment (kanemathers) [inline telnyx/api_resources/phone_number.py]: The /voice and /messaging nested URLs don't match our assumed pattern of [`/v2/:record_type/:id`](https://github.com/team-telnyx/telnyx-python/blob/8c0248523036bbd28d7e27a4ad201ef34afee484/telnyx/api_resources/abstract/api_resource.py#L45). I've had to overwrite a few methods to fix this.

File: telnyx/api_resources/phone_number.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    DeletableAPIResource,
+    ListableAPIResource,
+    UpdateableAPIResource,
+    nested_resource_class_methods,
+)
+from telnyx.six.moves.urllib.parse import quote_plus
+
+
+@nested_resource_class_methods("voice", operations=["list"], pluralize_path=False)
+@nested_resource_class_methods("voice", operations=["update"], pluralize_path=False)
```
Comment (nicktimko) [inline telnyx/api_resources/phone_number.py]: If `operations` is a list, can you combine them like
```suggestion
@nested_resource_class_methods("voice", operations=["list", "update"], pluralize_path=False)
```
?

Comment (kanemathers) [reply telnyx/api_resources/phone_number.py]: Good catch. I originally listed these out in order of the API documentation so I would be sure I didn't miss an endpoint. Forgot to clean it up.

File: telnyx/api_resources/abstract/nested_resource_class_methods.py [hunk 3]
```diff
@@ -4,9 +4,17 @@
 from telnyx.six.moves.urllib.parse import quote_plus
 
 
-def nested_resource_class_methods(resource, path=None, operations=None):
+def nested_resource_class_methods(
+    resource, path=None, operations=None, pluralize_path=True
+):
+    def make_path(v):
```
Comment (luizanao) [inline telnyx/api_resources/abstract/nested_resource_class_methods.py]: Will this work for an resource ended with `y`?
E.g: `emergency -> emergencys` ❌  ?

Comment (kanemathers) [reply telnyx/api_resources/abstract/nested_resource_class_methods.py]: No. The `nested_resource_class_methods` function didn't take much into account when it came to assuming an `s` suffix for method names.

I'll keep the function as it is, for now, and tackle this issues as they arise.

**Pull request merged**